Extension/readme.developer.md

```diff
@@ -0,0 +1,338 @@
+repository. This is done automatically when running the `yarn code`, but can be run manually.
+
+---
+### `yarn generate-native-strings`
+> #### `yarn generate-native-strings` - generates the native strings 
+
+This used to generate nativeStrings.ts and localized_string_ids.h from ./src/nativeStrings.json
+If adding localized strings to the native side, start by adding it to nativeStrings.json and use this to generate the others.
+
+> #### Note
+> The use of the `.scripts/common.ts:write()` function ensures that if the contents don't change the file won't be touched
+
+---
+### `yarn generate-options-schema`
+> #### `yarn generate-options-schema` - generates the options schema
+
+Inserts the options schema into `package.json` from the `tools/OptionsSchema.json` and the `tools/VSSymbolSettings.json` file.
+
+> #### Note
+> The use of the `.scripts/common.ts:write()` function ensures that if the contents don't change the file won't be touched
+
+---
+### `yarn copy-walkthrough-media`
+> #### `yarn copy-walkthrough-media` - copies the walkthrough media
+
+This copies the walkthrough media into the `dist` folder.
+
+> #### Note
+> The use of the `.scripts/common.ts:updateFiles()` function ensures that if the contents don't change the files won't be touched
+
+> #### `yarn copy-walkthrough-media watch` - watches for changes and automatically copies the walkthrough media
+
+This will watch for changes to the walkthrough files and automatically copy them across when they change.
+
+---
+### `yarn prep`
+> #### `yarn prep` - extension preparation
+
+This will call `yarn copy-walkthrough-media`, `yarn generate-native-strings`, `yarn translations-generate` to ensure
+that the generated files are up to date.
+
+---
+### `yarn lint`
+> #### `yarn lint` - lint all the source files (tests, ui, src, and .scripts)
+
+This will lint all the source files and report errors.
+
+> #### `yarn lint --fix` - lint all the source files and auto fix anything it can.
+
+---
+### `yarn compile`
+> #### `yarn compile` - compile the typescript source code
+
+Runs the typescript compiler on the source code (`test`,`src`,`ui`) and the output is placed in the `dist` folder.
+
+This means we don't use webpack for day-to-day use anymore. 
+
+This will also verify that the repository is prepped and tries to prep it if it's not.
+
+---
+### `yarn watch`
+> #### `yarn watch` - compile the source code and watch for changes
+
+Runs the typescript compiler in watch mode on the source code (`test`,`src`,`ui`) and the output is placed in the `dist` folder.
+
+Any changes to the source files will be automatically compiled when saved.
+
+This will also verify that the repository is prepped and tries to prep it if it's not.
+
+---
+
+### `yarn verify`
+> #### `yarn verify [--verbose]` - verifies that the repository is built correctly
+
+Checks for the presence of the compiled files and the prepped files ( see [`yarn verify prep`](#yarn-verify) )
```

Comment:
Why is there a space after `(`? I have not seen that done before. Is there a reason for that? Is this some rule I'm not aware of?